### DISCUSSION TITLE
Can this be used for template matching for finding similar objects

### DISCUSSION THREAD

**Discussion by ogencoglu** [Vincentqyw/image-matching-webui#102] [General]
I have several small template images and want to find the bounding boxes of similar objects in a large image. So it is not exactly trying to register an an image on top of another. 

Can any of the algorithms be suitable for this task? How can the matched keypoints be turned into bounding boxes of the found similar objects?

@Vincentqyw

**Comment by Vincentqyw** [comment]
Your task might be tackled using detection and matching.

Simply, detect multiple targets in small template images, yielding bounding boxes. For each box, use its features to match with the large image using image matching and RANSAC. If sufficient matches are found, compute the homography matrix to map the box to the large image. Repeat for all boxes.

However, issues may arise, such as significant scale changes between images requiring scale-invariant features, and multiple similar targets in the larger image may cause incorrect matches.

**Reply by ogencoglu** [reply]
Thanks for the reply.

> Simply, detect multiple targets in small template images, yielding bounding boxes.

Sorry, I don't follow. Let's say I have
- a single small image representing a door in floorplans = patch
- Large floorplans with lots of doors. But these doors are not exact pixel-perfect copy-paste of the patch (it is not a template-matching problem). They are not scaled/resized versions of the patch either. They are similar looking doors. Each floorplan might have slightly different door style. Doors can be in arbitrary orientation of course.

Can I detect these doors without having to train a standard object detection model?
